create ShuntingProblemSolutionValidator_09_18_2024.yp as the Solution Validator

asu-trans-ai-lab opened the pull request:
This commit introduces the ShuntingProblemSolutionValidator_09_18_2024.py, a python based solution validator for the shunting problem. The validator is designed to process input JSON files containing train movement and yard data, compute solution metrics such as track occupancy, movement time, and total shunting time, and validate them against the problem parameters. The script implements key data structures (TrainMovement, Solution, Input) and algorithms for movement (pull/push) validation, and outputs results to a CSV file. Additionally, the implementation includes:

Calculation of shunting times and track movements. Validation of train movements (pull and push) with reverse movement checks. Storage and handling of shunter and track occupancies. Output solution metrics for further analysis.

This Python code tool is now aligned with the sample spreadsheet provided in the sample_solution folder.

Commit:
create ShuntingProblemSolutionValidator_09_18_2024.yp as the Solution Validator

This commit introduces the ShuntingProblemSolutionValidator_09_18_2024.py, a python based solution validator for the shunting problem. The validator is designed to process input JSON files containing train movement and yard data, compute solution metrics such as track occupancy, movement time, and total shunting time, and validate them against the problem parameters. The script implements key data structures (TrainMovement, Solution, Input) and algorithms for movement (pull/push) validation, and outputs results to a CSV file. Additionally, the implementation includes:

Calculation of shunting times and track movements.
Validation of train movements (pull and push) with reverse movement checks.
Storage and handling of shunter and track occupancies.
Output solution metrics for further analysis.
This Python code tool is now aligned with the sample spreadsheet provided in the sample_solution folder.

MarcMeketonVillanova commented on solution_verification_code/ShuntingProblemSolutionValidator_09_18_2024.yp.txt:
```diff
@@ -0,0 +1,215 @@
+    num_wagons: int
+    track: int
+    TRK: List[List[int]] = field(default_factory=list)
+    Shunter: List[int] = field(default_factory=list)
```
Please use my latest code that has the length of the track being used and a column to show if any track is violating the track length constraint.

Commit:
Minor enhancement, use py as file extension.

Marc made some changes to the code: added N+1 columns, N=# of tracks.  N columns has the number of meters being used on a track.  The other column has a list of tracks that violate the length.

asu-trans-ai-lab commented:
I use .py as the file extension now.

MMeketon reviewed (approved):
Looks good.
